cmd/kube-proxy/app/server.go

```diff
@@ -258,6 +261,12 @@ func newProxyServer(ctx context.Context, config *kubeproxyconfig.KubeProxyConfig
 		return nil, err
 	}
 
+	if utilfeature.DefaultFeatureGate.Enabled(zpagesfeatures.ComponentFlagz) {
+		if flagzReader != nil {
+			s.flagz = flagzReader
+		}
+	}
```

Comment:
You don't need to check the feature gate here; `flagzReader` can only have been set if the feature gate is enabled.

(So you don't even need this as a separate section; you can just unconditionally set `flags: flagzReader` in the initial creation of the `ProxyServer` at the top of the function)

Reply:
Updated. thanks!